### PR TITLE
fix: Remote Online Store Type Inference Error with All-NULL Columns

### PR DESCRIPTION
**Pull request by yuan1j**
# What this PR does / why we need it

## 📋 Problem Description

When reading features from a **Feast Remote Online Store**, a `TypeError` is raised if an entire feature column contains only `NULL/NaN` values. The error occurs in the `python_values_to_proto_values()` function when it cannot infer the value type from empty data.

---

## 🔍 Root Cause Analysis

| Component | Issue |
|-----------|-------|
| **Location** | `feast/type_map.py` - `python_values_to_proto_values()` function |
| **Trigger** | Entire feature column contains `NULL/NaN` values |
| **Reason** | Type inference fails when all values are empty, resulting in `UNKNOWN` value type |
| **Current Behavior** | Raises `TypeError: Couldn't infer value type from empty value` |
| **Impact** | Remote online feature reading fails completely |

---

## 📊 Test Data Example

```
                   event_timestamp          driver_id  conv_rate  acc_rate  avg_daily_trips
0  2026-03-04 16:17:44+00:00       1001        NaN    1.000000           1000.0
1  2026-02-17 16:00:00+00:00       1005        NaN    0.708060            395.0
2  2026-02-17 17:00:00+00:00       1005        NaN    0.451616            145.0
```

> **⚠️ Note:** The `conv_rate` column contains all `NaN` values.

---

## 💡 Solution

Modify the type inference logic to handle `UNKNOWN` value types gracefully:

### Key Changes

1. ✅ **Do not raise error** when value type is `UNKNOWN`
2. ✅ **Preserve schema-defined type** from FeatureView definition
3. ✅ **Return proper NULL values** in the response while maintaining type information

### Code Change Summary

```python
# Before
if value_type is UNKNOWN:
    raise TypeError("Couldn't infer value type from empty value")

# After
# Remove this part code, and use schema-defined type or skip type inference for NULL-only columns
```

---

## ✨ Expected Behavior

After this fix:

- ✅ Remote online store can read columns with all-NULL values
- ✅ Feature schema type is preserved from FeatureView definition
- ✅ NULL values are returned correctly to the client
- ✅ No breaking changes to existing functionality

---

# Which issue(s) this PR fixes

Fixes: **Remote Online Store Type Inference Error with All-NULL Columns**

Related Error:
```
 Traceback (most recent call last):
  File "C:\svn\project\feature_store\feast\remote\test_workflow.py", line 130, in <module>
    run_demo()
  File "C:\svn\project\feature_store\feast\remote\test_workflow.py", line 25, in run_demo
    fetch_online_features(store)
  File "C:\svn\project\feature_store\feast\remote\test_workflow.py", line 121, in fetch_online_features
    returned_features = store.get_online_features(
  File "C:\Users\j\AppData\Local\Programs\Python\Python310\lib\site-packages\feast\feature_store.py", line 2445, in get_online_features
    response = provider.get_online_features(
  File "C:\Users\j\AppData\Local\Programs\Python\Python310\lib\site-packages\feast\infra\passthrough_provider.py", line 251, in get_online_features
    return self.online_store.get_online_features(
  File "C:\Users\j\AppData\Local\Programs\Python\Python310\lib\site-packages\feast\infra\online_stores\online_store.py", line 199, in get_online_features
    read_rows = self.online_read(
  File "C:\Users\j\AppData\Local\Programs\Python\Python310\lib\site-packages\feast\infra\online_stores\remote.py", line 169, in online_read
    message = python_values_to_proto_values(
  File "C:\Users\j\AppData\Local\Programs\Python\Python310\lib\site-packages\feast\type_map.py", line 838, in python_values_to_proto_values
    raise TypeError("Couldn't infer value type from empty value")
TypeError: Couldn't infer value type from empty value
```

---

# Misc

### Manual Test Steps

```bash
# 1. Start Feast services
cd C:\svn\project\feature_store\feast\test\feature_repo
feast ui
feast serve_registry
feast serve_offline
feast serve

# 2. Prepare test data with all-NULL column
# (conv_rate column contains only NaN values)

# 3. Run remote client
python remote/test_workflow.py
```

### Configuration Used

```yaml
project: test
registry: 
  registry_type: remote
  path: 127.0.0.1:6570
provider: local
online_store:
  type: remote
  path: http://127.0.0.1:6566
offline_store:
  type: remote
  host: 127.0.0.1
  port: 8815
entity_key_serialization_version: 3
auth:
  type: no_auth
```

---

## 📝 Additional Notes

| Item | Details |
|------|---------|
| **Breaking Changes** | None |
| **API Changes** | None |
| **Documentation** | May need update for NULL handling behavior |
| **Unit Tests** | Should add test case for all-NULL column scenario |
| **Affected Components** | `feast/type_map.py`, `feast/infra/online_stores/remote.py` |

---

## 👥 Reviewers

- @feast-maintainers
- @feast-infra-team

---

## ✅ Checklist

- [ ] Code follows Feast code conventions
- [ ] Unit tests added/updated
- [ ] Integration tests pass
- [ ] Documentation updated (if needed)
- [ ] Commits signed
- [ ] PR title follows conventional commits format

---

<a href="https://app.devin.ai/review/feast-dev/feast/pull/6063" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open with Devin
  </picture>
</a>

### CODE REVIEW

**Comment by yuan1j** [comment]
小军已收到

**Review by yuan1j — COMMENTED** [review]
Thank you for your helpful advice. This solution works for my case as well. Could you please review and approve the merge?

File: sdk/python/feast/type_map.py [hunk 1]
```diff
@@ -1015,7 +1015,7 @@ def python_values_to_proto_values(
             value_type = python_type_to_feast_value_type("", sample)
 
     if value_type == ValueType.UNKNOWN:
-        raise TypeError("Couldn't infer value type from empty value")
+        return [ProtoValue() for _ in values]
```
[the remark this answers is not in the document]
Comment (jyejare) [reply sdk/python/feast/type_map.py]: @yuan1j Looks like a perfect issue identification here. Could you please change the code to match the devins expectations here.

**Comment by yuan1j** [comment]
小军已收到

**Comment by yuan1j** [comment]
How can I merge these changes into the master branch?
